lightning/src/offers/merkle.rs

```diff
@@ -144,7 +145,7 @@ fn tagged_branch_hash_from_engine(
 /// [`Iterator`] over a sequence of bytes yielding [`TlvRecord`]s. The input is assumed to be a
 /// well-formed TLV stream.
 #[derive(Clone)]
-pub(super) struct TlvStream<'a> {
+pub(crate) struct TlvStream<'a> {
```

Comment:
I tried to reuse these `TlvStream/TlvRecord` structs since it seemed I was going to have to implement something similar. Should I move this out to some other file since or is it fine here?